client/scripts/views/components/search_bar.ts

```diff
@@ -225,295 +269,352 @@ const getBalancedContentListing = (unfilteredResults: any[], types: SearchType[]
+    filterMenuActive: boolean;
+    setUsingFilterMenu: Function;
+    searchQuery: SearchQuery;
+  }
+> = {
   view: (vnode) => {
     if (!vnode.state.searchTerm) vnode.state.searchTerm = '';
+    if (!vnode.state.searchQuery) vnode.state.searchQuery = new SearchQuery('', { isSearchPreview: true });
+    if (vnode.state.searchQuery.searchTerm !== vnode.state.searchTerm && vnode.state.searchTerm.length > 3) {
+      vnode.state.searchQuery.searchTerm = vnode.state.searchTerm
+    }
+    const { results, searchTerm, searchQuery } = vnode.state;
+    const isMobile = window.innerWidth < 767.98;
+
+    const setFilterMenuActive = (using: boolean) => {
+      vnode.state.filterMenuActive = using
+    }
+
+    vnode.state.closeResults = () => {
+      vnode.state.hideResults = true;
+    };
 
-    const { results, searchTerm } = vnode.state;
-    const showDropdownPreview = !m.route.get().includes('/search?q=');
-    const isMobile = (window.innerWidth < 767.98);
-    const LoadingPreview = m(List, {
-      class: 'search-results-loading'
-    }, [ m(ListItem, { label: m(Spinner, { active: true }) }) ]);
-    const searchResults = (!results || results?.length === 0)
-      ? (app.searchCache[searchTerm]?.loaded)
-        ? m(List, [ m(emptySearchPreview, { searchTerm }) ])
-        : LoadingPreview
-      : vnode.state.isTyping
-        ? LoadingPreview
-        : m(List, { class: 'search-results-list' }, results);
-    vnode.state.closeResults = () => { vnode.state.hideResults = true; };
+    vnode.state.setUsingFilterMenu = using => {
+      vnode.state.filterMenuActive = using
+    }
+
+    const executeSearch = (query: SearchQuery) => {
+      if (
+        !query.searchTerm ||
+        !query.searchTerm.toString().trim() ||
+        !query.searchTerm.match(/[A-Za-z]+/)
+      ) {
+        notifyError('Enter a valid search term');
+        return;
+      }
+      if (query.searchTerm.length < 4) {
+        notifyError('Query must be at least 4 characters');
+      }
+      query.isSearchPreview = false;
+      app.search.addToHistory(query)
+      m.route.set(`/search?${query.toUrlParams()}`);
+    }
+
+    const historyList = app.search.getHistory()
+      .map(h => getSearchHistoryPreview(h, setFilterMenuActive, executeSearch))
+
+
+    if(historyList.length > 0) {
+      historyList.push(
+        m(ListItem, {class: 'search-history-no-results upper-border',
+          label: 'Tip: You can use operators like \'single quotes\', and the keyword "or" to limit your search!'})
+      )
+    }
+
+    const activeCommunity = app.community ? app.community.name : vnode.state.searchQuery.communityScope
+    const activeChain = app.activeChainId() || vnode.state.searchQuery.chainScope
+    const scopeTitle = m(ListItem, {class: 'disabled', label: 'Scope'})
+
+    const scopeButtons = [SearchScope.Threads, SearchScope.Replies, SearchScope.Communities, SearchScope.Members]
+      .map(s => {
+        return m(Button, {
+          size: Size.LG,
```

Comment:
Try something like DEFAULT or SM size here, might be a better fit.

Reply:
Restyled!